W-13505366-aws-secrets-manager-kt (#2322)
mulesoft/docs-connectors: amazon-secrets-manager-properties-provider/1.0/modules/ROOT/pages/index.adoc

> = Amazon Secretes Manager Properties Provider 1.0 - Mule 4

Reviewer: ```suggestion
= Amazon Secrets Manager Properties Provider 1.0 - Mule 4
```

> Amazon Secretes Manager Properties Provider is a configuration that enables you to retrieve your secrets from Amazon Secrets Manager cloud service before your Mule app runs.

Reviewer: ```suggestion
Amazon Secrets Manager Properties Provider is a configuration that enables you to retrieve your secrets from Amazon Secrets Manager cloud service before your Mule app runs.
```

For information about compatibility and fixed issues, refer to the Amazon Secrets Manager Properties Provider Release Notes.

== Before You Begin

To use this properties provider, you must be familiar with:

* AWS Secrets Manager
* Anypoint Connectors
* Mule runtime engine (Mule)
* Elements and global elements in a Mule flow
* How to create a Mule app using Anypoint Studio

Before creating an app, you must have:

* Mule Enterprise Edition
* Credentials to access the AWS Secrets
* Anypoint Platform
* Anypoint Studio version 7.3.5 or later

== Common Use Cases for the Properties Provider

With AWS Secrets Manager Properties Provider, you can:

> * Configure values from AWS secrets manager for Mule applications.

Reviewer: ```suggestion
* Configure values from AWS Secrets Manager for Mule applications.
```

> * Retrieve secrets for SaaS applications as part of the trigger or action.

Reviewer: ```suggestion
* Retrieve secrets for SaaS applications as part of the source or operation.
```

* Specify the name of the secret to retrieve.

> * Authenticate to AWS using the available options from the AWS SDK IAM, default provider chain or Basic Authentication.

Reviewer: ```suggestion
* Authenticate to AWS using the available options from the AWS SDK IAM, default provider chain, or Basic Authentication.
```

* Specify the AWS server or region to connect to the appropriate server for your business unit.
* Be notified about errors while connecting to AWS.

> * Be notified about errors while retrieving secretes from AWS.

Reviewer: ```suggestion
* Be notified about errors while retrieving secrets from AWS.
```

== Connection Types

> Connectors can use the Amazon Secrets Manager Properties Provider in their configurations. For example, in the *Global Element Properties* window of Anypoint Studio, you can configure the following fields for the *AWS Secrets Manager Properties Override Config*:

Reviewer: ```suggestion
Connectors can use Amazon Secrets Manager Properties Provider in their configurations. For example, in the *Global Element Properties* window of Anypoint Studio, you can configure the following fields for the *AWS Secrets Manager Properties Override Config*:
```

[%header,cols="20s,25a,30a,15a,10a"]
|===
| Name | Type | Description | Default Value | Required

> | AWS Secrets Manager Region a| String |  AWS Secrets Manager region, for example `us-east-2`. |  | x

Reviewer: ```suggestion
| AWS Secrets Manager Region a| String |  AWS Secrets Manager region, for example, `us-east-2`. |  | x
```

| AWS Access Key a| String |  AWS access key ID. |  | 
| AWS Secret Key a | String |  AWS secret key ID. |  | 

> |Role ARN a| String | The Role ARN unique identifies a role to gain cross account access. |  |

Reviewer: ```suggestion
|Role ARN a| String | Uniquely identifies a role to gain cross-account access. |  |
```

|===

image::amazon-secrets-manager-properties.png[Example of Amazon Secrets Manager Properties Provider in Global Element Properties window of Anypoint Studio]

In the *Configuration XML* editor, the configuration looks like this:

[source,xml,linenums]
----
<aws-secrets-manager-properties-override:config name="AWS_Secrets_Manager_Properties_Override_Config" doc:name="AWS Secrets Manager Properties Override Config" >
		<aws-secrets-manager-properties-override:basic-connection region="us-east-1" />
		<aws-secrets-manager-properties-override:role-connection roleARN="arn:aws:iam::480410955647:role/AmazonSSMRoleForAutomationAssumeQuickSetup" />
	</aws-secrets-manager-properties-override:config>
----

> If you do not provide credentials, the default AWS connection follows the https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/credentials-chain.html[default credentials provider chain - AWS SDK for java 2.x^]

Reviewer: ```suggestion
If you do not provide credentials, the default AWS connection follows the https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/credentials-chain.html[default credentials provider chain - AWS SDK for Java 2.x^].
```

== See Also

> 

Reviewer: an optional suggestion: standardize the terminology of the name, like replace AWS Secrets Manager with Amazon Secrets Manager
